### PR TITLE
Fixed translation bug when the locale contained the region

### PR DESCRIPTION
**Pull request by Velin92**
Also added some tests to prevent the issue from happening again... and the italian translation

### CODE REVIEW

File: UnitTests/Sources/LocalizationTests.swift [hunk 1]
```diff
@@ -19,87 +19,77 @@ import XCTest
...
-        XCTAssertEqual(L10n.inviteFriendsText("Element", "element.io"), "Yalnızca Söz ve Anahtar Kelimeler")
+        XCTAssertEqual(L10n.testLanguageIdentifier, "it")
     }
 
     /// Test fallback language for a language not supported at all
-    func disabled_testFallbackOnNotSupportedLanguage() {
-        //  set app language to something Element don't support at all (chose 'Malay' language)
-        Bundle.elementLanguage = "ms"
+    func testStripRegionIfRegionalTranslationIsNotAvailable() {
+        //  set app language to something that includes also a region (it-IT)
```
Comment (stefanceriu) [inline UnitTests/Sources/LocalizationTests.swift:36]:
```suggestion
        // set app language to something that includes also a region (it-IT)
```

File: UnitTests/Sources/LocalizationTests.swift [hunk 2]
```diff
@@ -19,87 +19,77 @@ import XCTest
 
 class LocalizationTests: XCTestCase {
```
Comment (stefanceriu) [inline UnitTests/Sources/LocalizationTests.swift:20]: Very nice 👌

File: ElementX/Sources/Generated/Strings+Untranslated.swift [hunk 3]
```diff
@@ -71,24 +71,12 @@ public enum UntranslatedL10n {
 
 extension UntranslatedL10n {
   static func tr(_ table: String, _ key: String, _ args: CVarArg...) -> String {
-    let languages = Bundle.preferredLanguages
```
Comment (stefanceriu) [inline ElementX/Sources/Generated/Strings+Untranslated.swift:74]: What's the new logic here, don't we need preferred languages anymore? If so shouldn't delete the implementation from the Bundle extension?

**Review by stefanceriu — APPROVED** [review]
Looks good to me but honestly I'm still a bit confused on how this actually works within the NSE context.

File: ElementX/Sources/Generated/Strings+Untranslated.swift (hunk 3 above)
Comment (Velin92) [reply ElementX/Sources/Generated/Strings+Untranslated.swift:74]: essentially Untranslated is just a collection of strings that are set in the "en" language, in fact if you check the project navigator other languages are not selected (if you select one Xcode will automatically create the collection of string, but we dont' need more than one for Untranslated ofc)
Before we were cycling through AAAAAALL the languages, but for Untranslated is not really needed to do all this shenanigans, so we just go straight to the point and always get the "en" strings inside Untranslated.

Also yeah we need preferred language we just use it for L10n instead for UntranslatedL10n.

Comment (stefanceriu) [reply ElementX/Sources/Generated/Strings+Untranslated.swift:74]: Oooh I just realised these are the untranslated ones, sorry, my bad 🤦

**Comment by Velin92** [comment]
> Looks good to me but honestly I'm still a bit confused on how this actually works within the NSE context.

@stefanceriu 
Yeah I'll explain it better in the other PR when is merged but essentially the `url(resource, extension)` function just gets any resource that is assigned to the target, so we can either assign the strings directly to the NSE target (by just adding their import in the target.yml), or we can get them from the Bundle.app. Better to discuss this in the other PR

**Comment by stefanceriu** [comment]
> Better to discuss this in the other PR

Fair but I think this is the way to go:
>or we can get them from the Bundle.app.

**Pull request merged**